cadc-access-control-server/src/main/java/ca/nrc/cadc/auth/AuthenticatorImpl.java

```diff
@@ -147,6 +149,18 @@ public void augmentSubject(final Subject subject)
             PluginFactory pluginFactory = new PluginFactory();
             UserPersistence userPersistence = pluginFactory.createUserPersistence();
             Principal ldapPrincipal = getLdapPrincipal(subject);
+
+            // Remove the HttpPrincipal in subject, if any
+            Set<Principal> toBeRemoved = new HashSet<Principal>();
```

Comment:
Does this work?  `toBeRemoved = subject.getPrincipals(HttpPrincipal.class)`

Reply:
Great idea! It does. Now we are back to a one liner.